scripts/dotnet.py

```diff
@@ -477,16 +478,28 @@ def install(
         dotnetInstallScriptPath
     ] if platform == 'win32' else [dotnetInstallScriptPath]
 
-    # Install Runtime/SDKs
-    for channel in channels:
-        cmdline_args = dotnetInstallInterpreter + [
+    # If Version is supplied, pull down the specified version
+
+    cmdline_args = dotnetInstallInterpreter + [
             '-InstallDir', install_dir,
-            '-Architecture', architecture,
-            '-Channel', channel,
+            '-Architecture', architecture
+    ]
+    if version is not None:
+        cmdline_args = cmdline_args + [
+            '-Version', version,
         ]
         RunCommand(cmdline_args, verbose=verbose).run(
             get_repo_root_path()
         )
+    else:
```

Comment:
can we use the same block and just wrap the conditional around putting `-Version` or `-Channel` in the args?

Reply:
It might be possible for users to specify `--channels master --version 2.2.1`. To support this scenario we cannot place the usage of both arguments under different path of the same if block.

Reply:
What does that mean? Doesn't a version number supersede a channel?

Reply:
The channels side runs on a list of channels, so I'm not sure how that would work.

Reply:
Version does supersede channel. But if multiple channels have been chosen (which our current scripting allows for?) we would run for each channel grab this version, which seems like something we aren't interested in doing (even though the later grabs will just say "such and such version is already installed")

Reply:
> What does that mean? Doesn't a version number supersede a channel?
This is the scenario:
```cmd
rem The user wants to run benchmarks against the latest master and version 2.0.0 versions of the CLI.
benchmarks_ci.py --channels master -Version 2.0.0
```

Reply:
Optionally, it could be `--dotnet-versions` (an array of versions), that you can use to compare preview4 vs preview5.